src/semaphore_internal.h

```diff
@@ -29,6 +29,13 @@
 
 struct dispatch_queue_s;
 
+#if USE_FUTEX_SEM
+typedef struct {
+	/* Low 32 bits are the count; high bits are the waiter count. */
+	volatile uint64_t dfx_data;
+} dispatch_futex_t;
+#endif  /* USE_FUTEX_SEM */
```

Comment:
That's my point .. OK .. I code it up, then we discuss based on the code.
Its easier that way and maybe we just misunderstand each other here.